engine/src/main/java/org/terasology/rendering/dag/nodes/AlphaRejectBlocksNode.java

```diff
@@ -0,0 +1,142 @@
+    @Override
+    public void initialise() {
+        super.initialise();
+        playerCamera = worldRenderer.getActiveCamera();
+        addDesiredStateChange(new SetViewportToSizeOf(READ_ONLY_GBUFFER));
+        addDesiredStateChange(new EnableMaterial(CHUNK_SHADER.toString()));
+        chunkShader = getMaterial(CHUNK_SHADER);
+    }
+
+    /**
+     * Renders the world's semi-transparent blocks, i.e. tree foliage and terrain plants.
+     * Does not render fully opaque blocks, i.e. the typical landscape blocks.
+     *
+     * Takes advantage of the two methods
+     *
+     * - WorldRenderer.increaseTrianglesCount(int)
+     * - WorldRenderer.increaseNotReadyChunkCount(int)
+     *
+     * to publish some statistics over its own activity.
+     */
+    @Override
+    public void process() {
+        PerformanceMonitor.startActivity("rendering/chunksAlphaReject");
+
+        final Vector3f cameraPosition = playerCamera.getPosition();
+
+        int numberOfRenderedTriangles = 0;
+        int numberOfChunksThatAreNotReadyYet = 0;
+
+        READ_ONLY_GBUFFER.bind();
+        chunkShader.setFloat("clip", 0.0f, true);
+        chunkShader.activateFeature(ShaderProgramFeature.FEATURE_ALPHA_REJECT);
+
+        playerCamera.lookThrough(); // TODO: remove. Placed here to make the dependency explicit.
+
+        while (renderQueues.chunksAlphaReject.size() > 0) {
+            RenderableChunk chunk = renderQueues.chunksAlphaReject.poll();
+
+            if (chunk.hasMesh()) {
+                final Vector3f chunkPosition = chunk.getPosition().toVector3f();
+                final Vector3f chunkPositionRelativeToCamera =
+                        new Vector3f(chunkPosition.x * ChunkConstants.SIZE_X - cameraPosition.x,
+                                chunkPosition.y * ChunkConstants.SIZE_Y - cameraPosition.y,
+                                chunkPosition.z * ChunkConstants.SIZE_Z - cameraPosition.z);
+
+                chunkShader.setFloat3("chunkPositionWorld",
+                        chunkPosition.x * ChunkConstants.SIZE_X,
+                        chunkPosition.y * ChunkConstants.SIZE_Y,
+                        chunkPosition.z * ChunkConstants.SIZE_Z,
+                        true);
+                chunkShader.setFloat("animated", chunk.isAnimated() ? 1.0f : 0.0f, true);
+
+                // Effectively this just positions the chunk appropriately, relative to the camera.
+                // chunkPositionRelativeToCamera = chunkCoordinates * chunkDimensions - cameraCoordinate
+                GL11.glPushMatrix();
+                GL11.glTranslatef(chunkPositionRelativeToCamera.x, chunkPositionRelativeToCamera.y, chunkPositionRelativeToCamera.z);
+
+                chunk.getMesh().render(ChunkMesh.RenderPhase.ALPHA_REJECT); // TODO: remove face culling from this method
+                numberOfRenderedTriangles += chunk.getMesh().triangleCount();
+
+                GL11.glPopMatrix(); // Resets the matrix stack after the rendering of a chunk.
+
```

Comment:
Thank you for the recommendation @indianajohn. Intriguing approach moving those methods to the Mesh interface. I guess one concern I have with this approach is that it makes Mesh objects less of data containers and more of active players in the rendering. On a gut level I'd feel more comfortable with Meshes being pure data and the renderer processing that data. But I haven't looked at the Mesh interface and perhaps it is already -not- like that.

Regarding the individual points:
1. I agree. Eventually I really hope for the camera class to become very lightweight, only storing data and implementing accessors. 
2. Over the two years I have been on the project there have been a number of discussions regarding taking advantage of other libraries (i.e. LWJGL 3.x) or higher OpenGL releases. My hope is that by making the renderer more understandable and more structured it will be easier for people to develop not only additional effects but also more modern renderers. So, I'm with you: we are only limited by manpower.
3. Sounds good to me.
4. I confirm that's the case (dag focusing on state changes), at least so far. With @tdgunes we discussed the possibility for the DAG to be the basis also for runtime generation of shaders but it still seems far out.

Again, thank you for your recommendations. Keep tuned for more commits.